src/utils/attachments.ts

```diff
@@ -0,0 +1,40 @@
+import { flatten } from "fp-ts/lib/Array";
+import { PrescriptionData } from "generated/backend/PrescriptionData";
+import { CreatedMessageWithContent } from "generated/io-api/CreatedMessageWithContent";
+import { toBarcode } from "./barcode";
+
+const MIME_TYPES = {
+  png: "image/png",
+  svg: "image/svg+xml"
+};
+
+const toBarcodeAttachments = async (name: string, value: string) => {
+  const errorOrBarcodes = await toBarcode(value);
+  return errorOrBarcodes.fold(
+    () => [],
+    barcodes => [
+      { name, content: barcodes.png, mime_type: MIME_TYPES.png },
+      { name, content: barcodes.svg, mime_type: MIME_TYPES.svg }
+    ]
+  );
+};
+
+export const getMessageWithAttachments = async (
+  message: CreatedMessageWithContent,
+  prescriptionData: PrescriptionData
+) => {
+  const nreArrays = await toBarcodeAttachments("nre", prescriptionData.nre);
```

Comment:
an optimization would be run these in paralles using `Promise.all` (but the best option is to work with Taks instead of promises and use sequence(task), see https://grossbart.github.io/fp-ts-recipes/#/async?a=work-with-a-list-of-tasks-in-parallel)

to do this 
- return the TaskEither from `toBarcode` (do not call `run`)
- fold the taskEither into a Task and return it from `toBarcodeAttachments`
- sequence Task in getMessageWithAttachments and return the result (do not call run or then, return the array of tasks)
- the controller will call `getMessageWithAttachments(...).map(ResponseSuccessJson).run()`

<-- always try to call `run` at the outermost level and work with Task / TaskEithers internally

Reply:
Actually by removing `run` from `toBarcode` it returns `Task<Either<Error, IBarcodeOutput>>`

Reply:
Edit: Nevermind I've refactored the code by change `fold` to `foldTaskEither` in `toBarcode` function